diff --git a/http_cache_control.module b/http_cache_control.module
index 3aa0f9a..644eb27 100644
--- a/http_cache_control.module
+++ b/http_cache_control.module
@@ -58,6 +58,14 @@ function http_cache_control_form_system_performance_settings_alter(&$form, FormS
     '#description' => t('The maximum time a 500 level response can be cached by proxies. Note: this is subject to Drupal being able to render the response.'),
   ];
 
+  $form['caching']['403_max_age'] = [
+    '#type' => 'select',
+    '#title' => t('403 cache maximum age'),
+    '#default_value' => $config->get('cache.http.403_max_age'),
+    '#options' => $form['caching']['page_cache_maximum_age']['#options'],
+    '#description' => t('The maximum time a 403 level response can be cached by proxies. This is to buffer origin from requests prior to content releases'),
+  ];
+
   $form['#submit'][] = 'http_cache_control_form_system_performance_settings_submit';
 }
 
@@ -71,5 +79,6 @@ function http_cache_control_form_system_performance_settings_submit(&$form, Form
     ->set('cache.http.404_max_age', $form_state->getValue('404_max_age'))
     ->set('cache.http.302_max_age', $form_state->getValue('302_max_age'))
     ->set('cache.http.301_max_age', $form_state->getValue('301_max_age'))
+    ->set('cache.http.403_max_age', $form_state->getValue('403_max_age'))
     ->save();
 }
diff --git a/src/EventSubscriber/CacheControlEventSubscriber.php b/src/EventSubscriber/CacheControlEventSubscriber.php
index fd5f13f..bb3e1af 100644
--- a/src/EventSubscriber/CacheControlEventSubscriber.php
+++ b/src/EventSubscriber/CacheControlEventSubscriber.php
@@ -19,7 +19,7 @@ class CacheControlEventSubscriber implements EventSubscriberInterface {
 
     $response = $event->getResponse();
 
-    if (!$response->isCacheable()) {
+    if (!$response->isCacheable() && $response->getStatusCode() !== 403) {
       return;
     }
 
@@ -38,6 +38,10 @@ class CacheControlEventSubscriber implements EventSubscriberInterface {
         $ttl = $config->get('cache.http.301_max_age', $max_age);
         break;
 
+      case 403:
+        $ttl = $config->get('cache.http.403_max_age', $max_age);
+        break;
+
       default:
         $ttl = $config->get('cache.page.max_age');
         break;
@@ -46,7 +50,7 @@ class CacheControlEventSubscriber implements EventSubscriberInterface {
     // Allow modules that set their own max age to retain it.
     // If a response max-age is different to the page max-age
     // then this suggests the max-age has already been manipulated.
-    if ($max_age != $config->get('cache.page.max_age')) {
+    if ($max_age != $config->get('cache.page.max_age') && $response->getStatusCode() !== 403) {
       $ttl = $max_age;
     }
     $response->setSharedMaxAge($ttl);
